crates/driver/src/tests/setup/mod.rs

```diff
@@ -1009,6 +1014,24 @@ impl Test {
     pub fn web3(&self) -> &web3::Web3<DynTransport> {
         &self.blockchain.web3
     }
+
+    pub fn notifications(&self) -> Vec<solver::dto::Notification> {
+        self.notifications
+            .try_iter()
+            .map(|value| {
+                serde_json::from_value(value.clone())
+                    .context(format!("Error parsing notification: {}", value))
+            })
+            .collect::<Result<Vec<_>, _>>()
+            .unwrap()
```

Comment:
nit: If you are going to unwrap anyway wouldn't it be more elegant to just unwrap the `from_value()` call? I think the information that is provided in the default error message should be enough to debug things.

Reply:
> the fail-fast .collect::<Result<Vec<_>, _>>() approach

Arguably both approaches would fail equally fast. Also we wouldn't have an early return.
Just a nit, though. So I don't really care. Just seemed unintuitive to me to jump through extra hoops just to crash and burn right after. 😅